intake_stac/catalog.py

```diff
@@ -33,12 +33,15 @@ def __init__(self, stac_obj, **kwargs):
         else:
             raise ValueError(
                 "Expected %s instance, got: %s"
-                % (type(self._stac_cls), type(stac_obj))
+                % (self._stac_cls, type(stac_obj))
             )
 
         metadata = self._get_metadata(**kwargs.pop("metadata", {}))
 
-        name = kwargs.pop("name", self._stac_obj.id)
+        try:
+            name = kwargs.pop("name", self._stac_obj.id)
+        except AttributeError:
+            name = str(type(self._stac_obj))
```

Comment:
@matthewhanson - here is another place where we were tripped up while using a `ItemCollection`. There is not an `id` attribute on the class object. Not sure if it is possible, but it would be nice if `ItemCollection` inherited from `satstac.Thing`.

Reply:
@jhamman In sat-stac an ItemCollection isn't really a STAC ItemCollection, it's more of a Single File STAC as per the extension:
https://github.com/radiantearth/stac-spec/tree/dev/extensions/single-file-stac

The idea behind this was that a self-contained STAC catalog would be like a regular STAC catalog, except as you pointed out there's no `id`, nor are there any other catalog fields:
https://github.com/radiantearth/stac-spec/blob/dev/catalog-spec/catalog-spec.md#catalog-fields

I've posted an issue for STAC recommending that all the Catalog fields get added to the Single File STAC extension:
https://github.com/radiantearth/stac-spec/issues/691